### DISCUSSION TITLE
Add arbitrary amount of axis

### DISCUSSION THREAD

**Discussion by pablo2909** [arogozhnikov/einops#277] [Q&A] [answered]
Hi,

I am wondering if there is a way of adding an arbitrary amount of axis to an array `X` based on an array `Z`?  I would like to do the following in an einops way:

```python
import jax.numpy as jnp

X = jnp.arange(10)
Z = jnp.arange(10*3*4).reshape(10,3,4)

X.reshape([-1] + [1]*(Z.ndim-1))
```

Thank you for any suggestions

**Comment by arogozhnikov** [comment, marked as answer]
Hi Paul, einops does not provide such tool, you can use some trick like:

```python
result = (X.T + Z.T).T
```

**Reply by pablo2909** [reply]
Thank you ! Do you have an idea why this works though ? and not just `X + Y`.

**Reply by arogozhnikov** [reply]
broadcasting can automatically add arbitrary number of dummy axes on the left (but not on the right)

**Reply by pablo2909** [reply]
did not know that, thank you ! :)
